src/Symfony/Component/Messenger/Transport/NullTransport.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * Transport that send nothing. Useful for testing purpose.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class NullTransport implements TransportInterface
+{
+    /**
+     * @var Envelope[]
+     */
+    private $sent = [];
+
+    /**
+     * Is the transport stopped?
+     *
+     * @var bool
+     */
+    private $stopped = false;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        if ($this->stopped) {
+            return;
+        }
+
+        foreach ($this->sent as $envelope) {
+            $handler($envelope);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stop(): void
+    {
+        $this->stopped = true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope): Envelope
+    {
+        $this->sent[] = $envelope;
```

Comment:
As we accumulate envelopes over time, do we need to implement `ResetableInterface` to flush the envelopes from one request to the next?

Reply:
@sroze, @fabpot I've implemented `Symfony\Contracts\Service\ResetInterface` and I now see that there is an `ack` an `reject` method in the Transport interface. Do you think we need to store which message has been ack / nack?